### PR TITLE
add auth cluster image and update workflow

### PR DESCRIPTION
**Pull request by lingdie**
### Summary
📦🚀🔐

This pull request refactors the deployment process of the auth service to use sealos, a tool for building and running cluster images. It updates the workflow, dependencies, and manifests of the service, and adds new files to support sealos features, such as image preloading, template variables, and ingress configuration. It also removes some unused files and simplifies the `deploy.yaml` file.

> _We're building cluster images with sealos_
> _We're copying files and setting envs_
> _We're applying manifests and ingress rules_
> _Heave away, me hearties, heave away_

### Walkthrough
*  Add a new environment variable `SEALOS_VERSION` to the workflow and replace the step of building the cluster image on another repository with a new step of building the cluster image using sealos ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-48839d65f7bfedd91312ee406fd94da103ede7d3c7e791eb2c1657f46afdea00R29), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-48839d65f7bfedd91312ee406fd94da103ede7d3c7e791eb2c1657f46afdea00L152-R188))
* Add a new file `authImages` to list the images required by the auth service and its dependencies ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-00a6cc2bc19dc5babddeac2674e8b1c1c0b415da652d13be8249549eb35a4ba2L1-R3))
* Add a new file `Kubefile` to define how to build the cluster image for the auth service using sealos ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-f45ddcd259e20afc98b04a7806c33e2c386b441b59d8ee46eb66fb662411a85eR1-R13))
* Add a new file `configmap.yaml.tmpl` to define the configmaps and secrets for the auth service and its dependencies using template variables ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd46b769fac2404aa9a40593170848c3f25b2a0f7fb8b4648c26ed76ac3a1c2aL1-R77))
* Add a new file `ingress.yaml.tmpl` to define the ingress rules for the auth service and casdoor using template variables ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-ce733d8928680686199ea6990536c9af2a0d85c8ef320a65e5d8cef55d84f77bL1-R74))
* Update the versions of some dependencies and indirect dependencies in the `go.mod` file of the auth service ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-18aaf99e177d9d3d6e5df73dea7667a8d9e24d1fc1902a2c8dd6254c11520c8cL7-R11), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-18aaf99e177d9d3d6e5df73dea7667a8d9e24d1fc1902a2c8dd6254c11520c8cL19-R24), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-18aaf99e177d9d3d6e5df73dea7667a8d9e24d1fc1902a2c8dd6254c11520c8cL37-R56))
* Remove the secret and configmap definitions for mysql and casdoor from the `deploy.yaml` file, since they are now in the `configmap.yaml.tmpl` file ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd2c9f86d940ff23d04ff9ab53d36959cfa4bf5d38379e4111be106f5c2b3ceaL16-L53))
* Update the namespace of the auth service manager role, role binding, service, and deployment from `auth-service-system` to `sealos` in the `deploy.yaml` file ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd2c9f86d940ff23d04ff9ab53d36959cfa4bf5d38379e4111be106f5c2b3ceaL255-R217), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd2c9f86d940ff23d04ff9ab53d36959cfa4bf5d38379e4111be106f5c2b3ceaL270-R232), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd2c9f86d940ff23d04ff9ab53d36959cfa4bf5d38379e4111be106f5c2b3ceaL279-R241), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-fd2c9f86d940ff23d04ff9ab53d36959cfa4bf5d38379e4111be106f5c2b3ceaL297-R259))
* Add a new file `README.md` to document how to build and run the cluster image for the auth service using sealos ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-75c9ba47031e514bd06984109f7dbb182546b482bf42ea5e9cb46c991d7fb4fdR1-R27))
* Delete the files `service/auth/deploy/clusterimage/README.md`, `service/auth/deploy/clusterimage/Kubefile`, and `service/auth/deploy/clusterimage/manifests/deploy.yaml.tmpl`, since they are no longer needed ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-951cac946149b021b1376a11d8d2f6705d2c6d9601fbd50d61c5e0ac2312d405), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-639bf05092ae6bff44eacc868ead4bb532c02e9d30362fe6b6b96701029039d2), [link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-7d26bdf945cd9e6e3ec721c5bef227a9b49cb3110107f62d1853c48e50c05837))
* Delete the file `service/auth/deploy/manifests/configmap.yaml`, since it is replaced by the `configmap.yaml.tmpl` file ([link](https://github.com/labring/sealos/pull/2981/files?diff=unified&w=0#diff-9994fd256703a1c376374793177c8a0b3efd9e8cb44d791bfb0d4aa361b42715))

### CODE REVIEW

File: .github/workflows/services.yml [hunk 1]
```diff
@@ -149,23 +150,40 @@ jobs:
...
-#          issue-number: 140
-#          repository: labring-actions/cluster-image
-#          body: |
-#            /imagebuild_apps sealos-${{ matrix.module }}-service dev
-#          token: "${{ secrets.GH_PAT }}"
+  build-cluster-image:
+    needs:
+      - image-build
+    runs-on: ubuntu-latest
+    strategy:
```
Comment (cuisongliu) [inline .github/workflows/services.yml:157]: if: ${{ (github.event_name == 'push') || (inputs.push_mage == true) }}

File: .github/workflows/services.yml [hunk 2]
```diff
@@ -149,23 +150,40 @@ jobs:
...
+      matrix:
+        module: [ auth ]
+    steps:
+      - name: Install sealos
+        uses: labring/sealos-action@v0.0.7
+        with:
+          sealosVersion: "4.1.7"
+
+      - name: Sealos login to ghcr.io
+        if: ${{ (github.event_name == 'push') || (inputs.push_mage == true) }}
```
Comment (cuisongliu) [inline .github/workflows/services.yml:167]: delete

File: .github/workflows/services.yml [hunk 3]
```diff
@@ -149,23 +150,40 @@ jobs:
...
+        run: |
+          sudo sealos login -u ${{ github.repository_owner }} -p ${{ secrets.GH_PAT }} --debug ghcr.io
+
+      - name: Build ${{ matrix.module }}-service cluster image
+        if: ${{ github.event_name == 'pull_request' || (github.event_name == 'push') || (inputs.push_mage == true)}}
```
Comment (cuisongliu) [inline .github/workflows/services.yml:172]: delete

**Comment by cuisongliu** [comment]
using sudu sealos

**Comment by cuisongliu** [comment]
add jobs

```
      - name: Expose git commit data
        uses: rlespinasse/git-commit-data-action@v1
```

File: .github/workflows/services.yml [hunk 4]
```diff
@@ -149,23 +150,40 @@ jobs:
...
+        working-directory: service/${{ matrix.module }}/deploy
+        run: |
+          IMAGE_NAME=ghcr.io/${github.repository_owner}/sealos-${{ matrix.module }}-service:dev
+          sudo sealos build -t IMAGE_NAME-amd64 --platform linux/amd64 -f Kubefile
+          sudo sealos build -t IMAGE_NAME-arm64 --platform linux/arm -f Kubefile
```
[the remark this answers is not in the document]
Comment (lingdie) [reply .github/workflows/services.yml:177]: done

File: .github/workflows/services.yml (hunk 3 above)
Comment (lingdie) [reply .github/workflows/services.yml:172]: done

File: .github/workflows/services.yml (hunk 1 above)
Comment (lingdie) [reply .github/workflows/services.yml:157]: done

File: .github/workflows/services.yml [hunk 5]
```diff
@@ -149,23 +150,40 @@ jobs:
...
+
+      - name: Push ${{ matrix.module }}-service cluster image
+        if: ${{ (github.event_name == 'push') || (inputs.push_mage == true) }}
+        working-directory: service/${{ matrix.module }}/deploy
+        run: |
```
[the remark this answers is not in the document]
Comment (lingdie) [reply .github/workflows/services.yml:181]: ok

File: service/auth/deploy/images/shim/authImages [hunk 6]
```diff
@@ -0,0 +1,4 @@
+bitnami/mysql:8.0.31
+casbin/casdoor:v1.302.0
+labring/docker-casbin:main
+ghcr.io/labring/sealos-auth-service:dev
```
Comment (cuisongliu) [inline service/auth/deploy/images/shim/authImages:4]: delete this image

File: .github/workflows/services.yml [hunk 7]
```diff
@@ -149,23 +149,54 @@ jobs:
...
+
+      - name: Install sealos
+        uses: labring/sealos-action@v0.0.7
+        with:
+          sealosVersion: "4.1.7"
+
+      - name: Build ${{ matrix.module }}-service cluster image
+        working-directory: service/${{ matrix.module }}/deploy
+        run: |
+          IMAGE_NAME=ghcr.io/${{ github.repository_owner }}/sealos-cloud-${{ matrix.module }}-service:${GIT_COMMIT_SHORT_SHA}
```
Comment (cuisongliu) [inline .github/workflows/services.yml:177]: sed  ghcr.io/labring/sealos-auth-service:dev/ghcr.io/labring/sealos-auth-service:commit

File: .github/workflows/services.yml [hunk 8]
```diff
@@ -149,23 +149,54 @@ jobs:
...
-#      - name: Build ${{ matrix.module }}-service image
-#        uses: peter-evans/create-or-update-comment@v1
-#        if: ${{ github.repository_owner == env.DEFAULT_OWNER }}
-#        with:
-#          issue-number: 140
-#          repository: labring-actions/cluster-image
-#          body: |
-#            /imagebuild_apps sealos-${{ matrix.module }}-service dev
-#          token: "${{ secrets.GH_PAT }}"
+  build-cluster-image:
```
Comment (cuisongliu) [inline .github/workflows/services.yml:152]: https://github.com/labring/sealos/pull/2981/files#diff-48839d65f7bfedd91312ee406fd94da103ede7d3c7e791eb2c1657f46afdea00R100

fix commit

File: service/auth/deploy/images/shim/authImages (hunk 6 above)
Comment (lingdie) [reply service/auth/deploy/images/shim/authImages:4]: sed this to commit id after build it.

**Pull request merged**